assignments/prototypes.js

```diff
@@ -10,38 +10,74 @@
   
 /*
   === GameObject ===
-  * createdAt
-  * name
-  * dimensions (These represent the character's size in the video game)
-  * destroy() // prototype method that returns: `${this.name} was removed from the game.`
+  *11 createdAt
+  * 11name
+  * 11dimensions (These represent the character's size in the video game)
+  * 11destroy() // prototype method that returns: `${this.name} was removed from the game.`
+  * 
+  * DONE
 */
+"use strict"
+function GameObject (attributes) {
+  this.createdAt = attributes.createdAt;
+  this.name = attributes.name;
+  this.dimensions = attributes.dimensions;
+}
+
+GameObject.prototype.destroy = function  () {
+  return (`${this.name} was removed from the game`)
+}
 
 /*
   === CharacterStats ===
-  * healthPoints
-  * takeDamage() // prototype method -> returns the string '<object name> took damage.'
-  * should inherit destroy() from GameObject's prototype
+  11* healthPoints
+  11* takeDamage() // prototype method -> returns the string '<object name> took damage.'
+  11* should inherit destroy() from GameObject's prototype
 */
 
+function CharacterStats (attribute) {
+  GameObject.call(this, attribute)
+  this.healthPoints = attribute.healthPoints;
+}
+
+CharacterStats.prototype = Object.create(GameObject.prototype)
+
+CharacterStats.prototype.takeDamage = function(){
+  return (`${this.name} took damage`)
+}
+
 /*
   === Humanoid (Having an appearance or character resembling that of a human.) ===
-  * team
-  * weapons
-  * language
-  * greet() // prototype method -> returns the string '<object name> offers a greeting in <object language>.'
-  * should inherit destroy() from GameObject through CharacterStats
-  * should inherit takeDamage() from CharacterStats
+ 11 * team
+ 11 * weapons
+  11* language
+  11* greet() // prototype method -> returns the string '<object name> offers a greeting in <object language>.'
+  *111 should inherit destroy() from GameObject through CharacterStats
+  1111* should inherit takeDamage() from CharacterStats
 */
+
+function Humanoid (attr) {
+  this.team = attr.team;
+  this.weapons = attr.weapons;
+  this.language = attr.language;
+  CharacterStats.call(this, attr)
+}
```

Comment:
You are going to want to make sure you do your this binding as the first thing in your constructor. This saves you from weird bugs when things get more complex, as well as recognize immediately that this is a child object